### PR TITLE
Clean up Swedish validations

### PR DESCRIPTION
**Pull request by greysteil**
Swedish length validations weren't taking advantage of the recent pseudo IBAN work. This PR:
- splits out Swedish validations into their own class
- distinguishes between SWIFT validations and local validations (the later being stronger)
- cleans up `SwedishDetailsConverter` slightly by ensuring it is only ever passed local details

### CODE REVIEW

**Comment by greysteil** [comment]
@matt-thomson - can you review?

File: lib/ibandit/swedish_details_converter.rb [hunk 1]
```diff
@@ -1,5 +1,15 @@
 module Ibandit
   class SwedishDetailsConverter
+    # Converts local Swedish details into SWIFT details.
+    #
+    # Local details can be provided as either:
+    # - branch_code: clearing number, account_number: serial number
+    # - banch_code: nil, account_number: #{clearing number}#{serial number}
```
Comment (matt-thomson) [inline lib/ibandit/swedish_details_converter.rb]: `branch_code`

File: lib/ibandit/swedish_details_converter.rb [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+    #
+    # The reverse conversion (extracting local details from SWIFT details) is
+    # not possible, since the clearing number cannot be derived. You should NOT
+    # pass this class a SWIFT account number, as it will not convert it to
+    # local details successfully.
```
Comment (matt-thomson) [inline lib/ibandit/swedish_details_converter.rb]: What will happen if you do pass it a SWIFT account number? Can we detect and handle this case?

File: lib/ibandit/iban.rb [hunk 3]
```diff
@@ -276,19 +276,47 @@ def supports_iban_determination?
     def valid_swedish_details?
       return true unless country_code == 'SE'
 
-      bank_details = {
-        bank_code: swift_bank_code,
-        account_number: swift_account_number
-      }
+      if branch_code
```
Comment (matt-thomson) [inline lib/ibandit/iban.rb]: There's an odd split of logic here - some of the validation logic is in the `IBAN` class, some is in the `SwedishDetailsValidator` class.  Given that the `IBAN` class is getting pretty big and has lots of responsibilities, feels like we should just be able to do:

```
def valid_swedish_details?
  return true unless country_code == 'SE'
  SwedishDetailsValidator.valid?(...)
end
```

File: lib/ibandit/swedish_details_validator.rb [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+        cleaned_account_number.length == length
+      end
+    end
+
+    ##########################
+    # Private helper methods #
+    ##########################
+
+    def self.bank_info_for_clearing_code(clearing_code)
```
Comment (matt-thomson) [inline lib/ibandit/swedish_details_validator.rb]: Pull these methods out to a new class, and use it from `SwedishDetailsConverter`?

File: lib/ibandit/local_details_cleaner.rb [hunk 5]
```diff
@@ -441,20 +441,19 @@ def self.clean_ro_details(local_details)
...
-        swift_bank_code:      bank_code,
-        swift_account_number: converted_details[:swift_account_number]
-      }
+      if local_details[:bank_code]
+        # If a bank_code was provided without a branch code we're (probably)
+        # dealing with SWIFT details and should just return them.
+        return local_details.merge(
+          swift_account_number: local_details[:account_number],
+          swift_bank_code: local_details[:bank_code]
+        )
```
Comment (matt-thomson) [inline lib/ibandit/local_details_cleaner.rb]: Is merging right here? If we think we've been given SWIFT details, then we probably shouldn't return any local details, as we can't deduce those from the SWIFT.

File: lib/ibandit/swedish_details_converter.rb (hunk 2 above)
Comment (greysteil) [inline lib/ibandit/swedish_details_converter.rb]: We can't detect - SWIFT details would look like just passing us an account number, but that's a legitimate argument to pass anyway. Best thing I could think of was this big warning.

In practice I don't think this should be a problem, given the checking we do at the call site (above).

File: lib/ibandit/iban.rb (hunk 3 above)
Comment (greysteil) [inline lib/ibandit/iban.rb]: Yeah, validation in this class feels like a larger problem. Didn't want to introduce inconsistency here so used `SwedishDetailsValidator` just for validation helper methods. Happy to revisit, but probably best to do in another PR?

File: lib/ibandit/swedish_details_converter.rb (hunk 2 above)
Comment (matt-thomson) [inline lib/ibandit/swedish_details_converter.rb]: Yep, should be fine.

File: lib/ibandit/iban.rb (hunk 3 above)
Comment (matt-thomson) [inline lib/ibandit/iban.rb]: Yep, agree.

**Comment by greysteil** [comment]
@matt-thomson - updated.

**Comment by matt-thomson** [comment]
:+1:
